File: 이진영/추가문제/23일차/풍선 터뜨리기.java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        String[] nums = br.readLine().split(" ");
        List<Integer> answer = new ArrayList<>();
        Deque<Integer[]> deq = new ArrayDeque<>();
        int count = 0;

        for (int i = 0; i < n; i++) {
            deq.add(new Integer[]{i + 1, Integer.parseInt(nums[i])});
        }

        while (deq.size() > 0) {
            while (count != 0) {
                if (count > 0) {
                    count--;
                    deq.add(deq.pollFirst());
                } else {
                    count++;
                    deq.addFirst(deq.pollLast());
                }
            }

            Integer[] cur = deq.pollFirst();
            count = cur[1] > 0 ? cur[1] - 1 : cur[1];
            answer.add(cur[0]);
        }

        System.out.println(answer
        .stream()
        .map(Object::toString)
        .collect(Collectors.joining(" ")));
    }
}

File: 이진영/추가문제/23일차/풍선 터뜨리기.js
const main = () => {
    const input = require('fs').readFileSync('dev/stdin')
    .toString().trim().split("\n").map(i => i.trim());
    
    let balloons = input[1].split(" ").map((v, i) => [i + 1, v]);
    let count = 0;
    let answer = [];
    
    while (balloons.length > 0) {
        while (count !== 0) {
            if (count > 0) {
                count--;
                balloons.push(balloons.shift());
            } else {
                count++;
                balloons.unshift(balloons.pop());
            }
        }

        let cur = balloons.shift();
        count = cur[1] > 0 ? cur[1] - 1 : cur[1];
        answer.push(cur[0]); 
    }

    console.log(answer);
}

main();